src/main/java/seedu/address/model/Entity/Entity.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.model.Entity;
+
+public abstract class Entity {
+    protected final ID id; // Note: id is unique and should never be mutable.
+    protected String name;
+
+    /**
+     * Constructor.
+     *
+     * @param id
+     * @param name
+     */
+    Entity(ID id, String name) {
+        this.id = id;
+        this.name = name;
+    }
+
+    // Getter methods
+
+    /**
+     * Gets the name.
+     *
+     * @return String
+     */
+    public String getName() {
+        return this.name;
+    }
+
+    /**
+     * Gets the id.
+     *
+     * @return ID
+     */
+    public ID getId() {
+        return this.id;
+    }
+
+    // Setter methods
+
+    /**
+     * Sets the name of the entity.
+     *
+     * @param name
+     */
+    public void setName(String name) {
+        this.name = name;
+    }
+}
```

Comment:
What's the difference between list and view? It's implemented on EntityList, but Entity should have nothing to list/view?

I'll add toString and Equals and Hash to all of them so that everyone can do simple equality check.

Reply:
From my understanding, list shows all entities with minimal info while view shows one entity with all its info right? I was thinking toString() can be used in listcommands while view() can be used in viewcommands.

Reply:
Hmm right, I totally forgot about this from our discussion. How about I name this as viewDetailed() and viewMinimal(). This will vary from entity to entity and will be called directly from the entity object? is that okay?

Reply:
Yup can do